packages/core/src/base/staticClient.ts

```diff
@@ -93,6 +93,30 @@ export const logResponse = (response: Response): void => {
   );
 };
 
+/**
+ * Exports a minipass-fetch Headers object as a map.
+ * 
+ * minipass-fetch Headers class has a function to convert a Headers object into
+ * a map. But that function creates an array of values for each header. 
+ * `http-cache-semantics`, a package used by `make-fetch-happen` to manipulate
+ * headers expects them to be comma separated strings.
+ *
+ * @param headers - a minipass-fetch Headers object
+ *
+ * @returns a map of headers
+ */
+export function exportHeadersAsMap(headers: Headers) {
```

Comment:
The function name name is inaccurate:
1) This function works with any iterable, not just headers.
2) This function returns an Object, not a Map.